utils/self-extracting-executable/post_build.sh

```diff
@@ -1,6 +1,10 @@
 padding="               "
 if [[ $OSTYPE == 'darwin'* ]]; then
-    sz="$(stat -f %z 'decompressor')"
+    if which gstat; then
```

Comment:
The GNU and the Mac version stat support different command line arguments (`-c %s` vs `-f %z`). On Mac, we are now checking for `gstat` which is the GNU version which can be installed via brew. Since installing `gstat` via brew does not remove or disable the build-in Mac version of `stat`, I wonder how this PR "fixes the macOS build" (commit message)?

Reply:
as you say, i installed gnu coreutils by homebrew in my macOS. and `gstat` is the default alternative for `stat`

Reply:
Instead of making the script more complex, we could continue to call `stat` even if the user installed `gstat`.

Reply:
the problem is directly call stat will fail, because it's actually 'gstat', the running params are different